### PR TITLE
Add ggplot fig

### LINKED ISSUE #20: Convert plot from base R to `ggplot`
The current plot uses base R. We should flip this into ggplot. At that time we can also double check the minimum viable version of targets (see [this](https://github.com/USGS-R/ds-pipelines-targets-1-course/pull/19#discussion_r1152742010) comment from the review of PR #19)

### PR DESCRIPTION
**Pull request by jpadilla-usgs**
## Overview

This PR is the first in a series of revamps to the targets training series. This first MR is motivated by switching the basic plot from base R to ggplot2. All code updates flow from that initial decision.

A successful merge of this PR will close #20

## Summary of major changes
* I converted the plotting code for `figure_1.R` from `base` to `ggplot2`
* I modified `eval_data` significantly to complete the first bullet
  * Originally `eval_data` was a subset of results from [`me_RMSE.csv`](https://www.sciencebase.gov/catalog/item/5d925066e4b0c4f70d0d0599) with three additional columns added: one for point/line color, one for point shape, and one for the number of profiles. `eval_data` was then used to make the plot, save the intermediary file (as `model_summary_results.csv`) and then populate a data summary (in `model_diagnostic_text.txt`)
  * My update to `eval_data` is still able to produce `figure_1.png` and `model_diagnostic_text.txt` (see Verification section); however, now `model_summary_results.csv` is a true summary of model stats rather than 
* I updated the code that is used to populate the model diagnostic summary to be able to use the updated version of `eval_data`

## Requested review
* Run `_my_work__R/my_happy_script.R` in its entirety and verify that you can successfully reproduce the following outputs in `my_dir`: `figure_1.png`, `model_diagnositc_text.txt`, and `model_summary_results.csv`

### Review of changes in `eval_data`
old:
```
> str(eval_data_old)
tibble [85 × 7] (S3: tbl_df/tbl/data.frame)
 $ exper_n   : int [1:85] 1 1 1 1 1 1 1 1 1 1 ...
 $ exper_id  : chr [1:85] "similar_10" "similar_10" "similar_10" "similar_100" ...
 $ model_type: chr [1:85] "dl" "pb" "pgdl" "dl" ...
 $ rmse      : num [1:85] 3.2 1.95 1.4 2.02 1.49 ...
 $ col       : chr [1:85] "#d95f02" "#1b9e77" "#7570b3" "#d95f02" ...
 $ pch       : int [1:85] 22 21 23 22 21 23 21 23 22 21 ...
 $ n_prof    : num [1:85] 10 10 10 100 100 100 2 2 50 50 ...```
```
new:
```
> str(eval_data_new)
gropd_df [17 × 10] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ exper_id    : chr [1:17] "similar_10" "similar_10" "similar_10" "similar_100" ...
 $ model_type  : Factor w/ 3 levels "pgdl","dl","pb": 2 3 1 2 3 1 3 1 2 3 ...
 $ mean        : num [1:17] 3.35 1.96 1.47 1.73 1.64 ...
 $ min         : num [1:17] 3.07 1.57 1.31 1.55 1.43 ...
 $ max         : num [1:17] 3.94 2.77 1.71 2.02 1.99 ...
 $ ct_exper    : int [1:17] 5 5 5 5 5 5 5 5 5 5 ...
 $ col         : chr [1:17] "#d95f02" "#1b9e77" "#7570b3" "#d95f02" ...
 $ pch         : num [1:17] 22 21 23 22 21 23 21 23 22 21 ...
 $ n_prof      : num [1:17] 10 10 10 100 100 100 2 2 50 50 ...
 $ model_legend: Factor w/ 3 levels "Process-Guided Deep Learning",..: 2 3 1 2 3 1 3 1 2 3 ...
```

### Figure side-by-side
image

### Template text side-by-side
image

## A few final thoughts on this training
* You'll notice that I removed helpful comments on purpose ([752057c](https://github.com/DOI-USGS/ds-pipelines-targets-1-course/pull/24/commits/752057c9db03faf2f4162c5e710c220ba723466e))\
* There are probably other refinements and improvements that could be made, but I wanted to keep the spirit of the code as close to the original as possible.

## Next steps
* Tackle the figures for repo structure and pick up conversation on .placeholder (maybe)
* Add more links tocode in the text

### CODE REVIEW

File: my_work_R/my_happy_script.R [hunk 1]
```diff
@@ -1,93 +1,101 @@
...
+        "Deep Learning" = "dl",
+        "Process-Based" = "pb"
+      )
+    )
+
+g <- 
+  ggplot(data = eval_data, 
+         aes(color = model_legend, shape = model_legend)) +
+  geom_line(aes(x = n_prof, y = mean_rmse), 
+            linetype = "dashed",
```
Comment (cnell-usgs) [inline my_work_R/my_happy_script.R:53]: I think the prior was "dotted" but either works

File: my_work_R/my_happy_script.R [hunk 2]
```diff
@@ -1,93 +1,101 @@
...
+    col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+      ), 
+    pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+      ),
```
Comment (cnell-usgs) [inline my_work_R/my_happy_script.R:36]: remove these variables, they aren't being used anymore

File: my_work_R/my_happy_script.R [hunk 3]
```diff
@@ -1,93 +1,101 @@
...
+  ggplot(data = eval_data, 
+         aes(color = model_legend, shape = model_legend)) +
+  geom_line(aes(x = n_prof, y = mean_rmse), 
+            linetype = "dashed",
+            lwd = 0.5,
+            position = position_dodge(width = 0.05)) +
+  geom_linerange(aes(x = n_prof, ymin = min_rmse, ymax = max_rmse),
+                 lwd = 0.5,
+                 position = position_dodge(width = 0.05)) +
+  geom_point(aes(x = c(2, 10, 50, 100, 500, 980), y = mean_rmse),
```
Comment (cnell-usgs) [inline my_work_R/my_happy_script.R:59]:
```suggestion
  geom_point(aes(x = n_prof, y = mean_rmse), 
```

File: my_work_R/my_happy_script.R [hunk 4]
```diff
@@ -1,93 +1,101 @@
...
+             position = position_dodge(width = 0.05), 
+             fill = "white",
+             size = 3) +
+  labs(
+    x = "Training Temperature Profiles (#)",
+    y = "Test RMSE (°C)"
+  )
+
+g <- g +
+  scale_x_log10(breaks = n_profs, limits = c(1, 1000)) +
```
Comment (cnell-usgs) [inline my_work_R/my_happy_script.R:69]:
```suggestion
  scale_x_log10(breaks = unique(eval_data$n_prof), limits = c(1, 1000)) +
```

Comment (cnell-usgs) [reply my_work_R/my_happy_script.R:69]: this was needed to make the plot code work

Comment (cnell-usgs) [inline my_work_R/my_happy_script.R:69]: Also suggest dropping the `limits` here to make the spacing between the y-axis and the chart elements a little closer

**Review by cnell-usgs — COMMENTED** [review]
figure_1

I was able to reproduce with the minor edit to the x-axis breaks. I left a few comments to clean up before merging.

File: my_work_R/my_happy_script.R (hunk 4 above)
Comment (jpadilla-usgs) [reply my_work_R/my_happy_script.R:69]: Fix manually

File: my_work_R/my_happy_script.R (hunk 3 above)
Comment (jpadilla-usgs) [reply my_work_R/my_happy_script.R:59]: Fixed manually

File: my_work_R/my_happy_script.R (hunk 1 above)
Comment (jpadilla-usgs) [reply my_work_R/my_happy_script.R:53]: resolved

File: my_work_R/my_happy_script.R (hunk 2 above)
Comment (jpadilla-usgs) [reply my_work_R/my_happy_script.R]: resolved

**Pull request merged**